### PR TITLE
docs: rework and expand docs for utilities package

### PR DESCRIPTION
**Pull request by hudlow**
## PR summary
Generate dedicated documentation for constants and functions available in the utilities package.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

### CODE REVIEW

**Comment by dpopp07** [comment]
> Should we also update the copyright prologs in the ruleset and validator packages so they're consistent across the project?

I haven't finished my review but I was going to ask about that, too.

Is there a reason they needed to change?

**Comment by hudlow** [comment]
@padamstx @dpopp07 this was all a part of running the JSDoc plugin for ESLint which, per an offline discussion with Dustin, I have decided to pull out of this PR. I also reverted those header changes and we can add them back if/when we enable JSDoc linting.

**Comment by padamstx** [comment]
I took a look at the latest changes and I still approve :smile:

File: scripts/generate-utilities-docs.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env node
+
+const mustache = require("mustache");
```
Comment (dpopp07) [inline scripts/generate-utilities-docs.js:3]: This JS style for this script should be consistent with the rest of this project (single quotes, trailing commas, etc.)

Is it included in the linting?

File: scripts/generate-utilities-docs.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+const fs = require("node:fs");
+const childProcess = require("node:child_process");
```
Comment (dpopp07) [inline scripts/generate-utilities-docs.js:5]: Optional nit: this looks fine but when internal Node modules are used elsewhere in the project, we drop the `node:` (which is optional) and use object deconstruction in the imports:

```suggestion
const { readFileSync, writeFileSync } = require('fs');
const { execSync } = require('child_process');
```

This is slightly more semantic than *purely* stylistic so again, I'm happy to keep it as is! Just wanted to make a note for consistency's sake.

File: scripts/generate-utilities-docs.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  })
+);
+
+const rendered = mustache.render(
+  template,
+  { package, constants, functions },
+);
+
+fs.writeFileSync(
+  "docs/utilities.md",
```
Comment (dpopp07) [inline scripts/generate-utilities-docs.js:60]: Another nit but connecting the doc with the NPM package may provide more clarity for readers
```suggestion
  'docs/openapi-ruleset-utilities.md',
```

File: packages/utilities/src/collections/index.js [hunk 4]
```diff
@@ -1,98 +1,102 @@
 /**
- * Copyright 2017 - 2024 IBM Corporation.
- * SPDX-License-Identifier: Apache2.0
+ * @file
+ * @copyright IBM Corporation 2017–2024
+ * @license Apache-2.0
```
Comment (dpopp07) [inline packages/utilities/src/collections/index.js:4]: This looks leftover from the previous changes that are now split out into #703 

I think it's worth keeping this commit clean

**Review by dpopp07 — CHANGES_REQUESTED** [review]
This looks awesome!

I left a few very, very minor comments that would be good to address before merging.

**Review by dpopp07 — APPROVED** [review]
Looks great! 👍

**Pull request merged**